### DISCUSSION TITLE
Open WebUI Fails to Connect to Local Ollama Instance After Update Without Ollama Running

### DISCUSSION THREAD

**Discussion by Foadsf** [open-webui/open-webui#2777] [General]
as a follow up to [this question](https://discord.com/channels/1121039057993089076/1131623060315844710/1247072529365991424):

## Description

**Bug Summary:**
After updating and running Open WebUI through Pinokio without running Ollama first, Open WebUI is no longer able to communicate with my local Ollama instance. This has caused the application to fail to start correctly, resulting in a black screen when accessing `http://127.0.0.1:8000/`.

**Steps to Reproduce:**
1. Install Open WebUI through [Pinokio](https://program.pinokio.computer/#/?id=install) and Ollama via winget.
2. Ensure Open WebUI is running smoothly.
3. Forget to start Ollama and update+run Open WebUI through Pinokio once.
4. Attempt to restart Open WebUI with Ollama running.
5. Observe the black screen and failure to connect to Ollama.

**Expected Behavior:**
Open WebUI should connect to Ollama and function correctly even if Ollama was not started before updating Open WebUI. Ideally, updating Open WebUI should not affect its ability to communicate with Ollama.

**Actual Behavior:**
Open WebUI fails to communicate with the local Ollama instance, resulting in a black screen and failure to operate as expected.

## Environment

- **Open WebUI Version:** `0.2.2`
- **Ollama:** `0.1.41`

- **Operating System:** Microsoft Windows [Version 10.0.19045.4412]
- **Browser:** Microsoft Edge

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
1. Open Microsoft Edge and press `F12` to open Developer Tools.
2. Go to the `Console` tab.
3. Clear any existing logs using the clear button.
4. Refresh the page and reproduce the issue.
5. Right-click within the Console window, select `Save as...`, and save the logs to a file.
6. Attach the saved file [here](https://github.com/user-attachments/files/15530382/127.0.0.1-1717400344351.log).

**Docker Container Logs:**
N/A

**Screenshot:**
![Pinokio_XF4NC8QgxF](https://github.com/open-webui/open-webui/assets/12762442/da0117f0-77d7-4a2f-b179-2ddd7ca885ad)

## Installation Method

Installed Open WebUI through Pinokio and Ollama via winget.

## Additional Information

I believe the issue arose because I forgot to start Ollama before updating Open WebUI through Pinokio. This might have changed some settings, preventing Open WebUI from communicating with Ollama.

The most important aspect for me is to restore my previous chats. I prefer not to reinstall Open WebUI, or if necessary, I need a way to back up my previous chats.

**Comment by phonosys** [comment]
I'm having the same issue, i don't think its related to Ollama's status (mine is running and working).

raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url=URL('https://api.openai.com/v1/api/models')

looks like error handling issue?

**Comment by Foadsf** [comment]
@phonosys, thanks for the input. I suspect the issue might be that Open WebUI is unable to communicate with Ollama correctly and defaults to the OpenAI API.

To troubleshoot, I want to:
1. Verify if Ollama is running and accessible. Check if it's exposed at the configured IP address and port.
2. Ensure the `config.yaml` or relevant configuration file in Open WebUI correctly points to Ollama.

If Ollama runs on a different port or IP than expected, I need to update the Open WebUI configuration to match. I'll also be sure to check the error handling as you suggested.

Any tips on verifying Ollama's status would be appreciated.

**Comment by phonosys** [comment]
On my end Ollama runs just fine, can run and switch models using the configured IP and port.
on the open-webui side:
empty model list (Open WebUI is unable to communicate with Ollama correctly like you mentioned)
INFO:apps.openai.main:get_all_models()
None

**Comment by Foadsf** [comment]
asked a new question [here](https://discord.com/channels/1170866489302188073/1247114046549921812/1247114046549921812) on Discord.

**Comment by davidlight2018** [comment]
Got same problems.

> On my end Ollama runs just fine, can run and switch models using the configured IP and port. on the open-webui side: empty model list (Open WebUI is unable to communicate with Ollama correctly like you mentioned) INFO:apps.openai.main:get_all_models() None

**Comment by Foadsf** [comment]
I ended up uninstalling Open WiebUI and reinstalling everything back. I lost all my previous information! 😞

**Comment by phonosys** [comment]
looks like its still broken on 0.2.3.
guess ill try to re-install.

**Comment by limpep** [comment]
Having same issue, switched to v0.1.125 and works fine

here is the error I am getting
```

INFO:     127.0.0.1:61987 - "GET /static/favicon.png HTTP/1.1" 200 OK
ERROR:apps.openai.main:404, message='Not Found', url=URL('https://api.openai.com/v1/api/models')
Traceback (most recent call last):
  File "C:\Users\USER\Documents\projects\ai\open-webui\backend\apps\openai\main.py", line 473, in proxy
    r.raise_for_status()
  File "C:\Python312\Lib\site-packages\aiohttp\client_reqrep.py", line 1070, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url=URL('https://api.openai.com/v1/api/models')
INFO:     127.0.0.1:61990 - "GET /openai/api/models HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:61996 - "GET /api/v1/modelfiles/ HTTP/1.1" 404 Not Found
ERROR:    Exception in ASGI application
  + Exception Group Traceback (most recent call last):
  |   File "C:\Python312\Lib\site-packages\starlette\_utils.py", line 87, in collapse_excgroups
  |     yield
  |   File "C:\Python312\Lib\site-packages\starlette\middleware\base.py", line 190, in __call__
  |     async with anyio.create_task_group() as task_group:
  |   File "C:\Python312\Lib\site-packages\anyio\_backends\_asyncio.py", line 678, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "C:\Python312\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 435, in run_asgi
    |     result = await app(  # type: ignore[func-returns-value]
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "C:\Python312\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 78, in __call__
    |     return await self.app(scope, receive, send)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "C:\Python312\Lib\site-packages\fastapi\applications.py", line 1054, in __call__
    |     await super().__call__(scope, receive, send)
    |   File "C:\Python312\Lib\site-packages\starlette\applications.py", line 123, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "C:\Python312\Lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    |     raise exc
    |   File "C:\Python312\Lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    |     await self.app(scope, receive, _send)
    |   File "C:\Python312\Lib\site-packages\starlette\middleware\base.py", line 189, in __call__
    |     with collapse_excgroups():
    |   File "C:\Python312\Lib\contextlib.py", line 158, in __exit__
    |     self.gen.throw(value)
    |   File "C:\Python312\Lib\site-packages\starlette\_utils.py", line 93, in collapse_excgroups
    |     raise exc
    |   File "C:\Python312\Lib\site-packages\starlette\middleware\base.py", line 191, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "C:\Users\neo\Documents\projects\ai\open-webui\backend\main.py", line 373, in update_embedding_function
    |     response = await call_next(request)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "C:\Python312\Lib\site-packages\starlette\middleware\base.py", line 166, in call_next
    |     raise RuntimeError("No response returned.")
    | RuntimeError: No response returned.
    +------------------------------------

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 435, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\fastapi\applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "C:\Python312\Lib\site-packages\starlette\applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Python312\Lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    raise exc
  File "C:\Python312\Lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "C:\Python312\Lib\site-packages\starlette\middleware\base.py", line 189, in __call__
    with collapse_excgroups():
  File "C:\Python312\Lib\contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "C:\Python312\Lib\site-packages\starlette\_utils.py", line 93, in collapse_excgroups
    raise exc
  File "C:\Python312\Lib\site-packages\starlette\middleware\base.py", line 191, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\Documents\projects\ai\open-webui\backend\main.py", line 373, in update_embedding_function
    response = await call_next(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\starlette\middleware\base.py", line 166, in call_next
    raise RuntimeError("No response returned.")
```

**Comment by adrum** [comment]
I'm running this natively on macOS, and experienced this after a git pull. Running `npm run build` from the root of the repo fixed the issue for me.

**Comment by bstotmeister** [comment]
I was also running into this issue after pulling the latest after not updating since late May/early June 2024.
I tried a few things above, but what ultimately resolved it for me was updating Ollama, then closing and restarting it.
